bundles/org.openhab.binding.smsmodem/src/main/java/org/openhab/binding/smsmodem/internal/handler/SMSModemBridgeHandler.java

```diff
@@ -0,0 +1,473 @@
+    }
+
+    private void checkParam(SMSModemBridgeConfiguration config) throws ModemConfigurationException {
+        String realSerialPort = resolveEventualSymbolicLink(config.serialPort);
+        SerialPortIdentifier identifier = serialPortManager.getIdentifier(realSerialPort);
+        if (identifier == null) {
+            // no serial port
+            throw new ModemConfigurationException(
+                    realSerialPort + " with " + config.baud + " is not a valid serial port | baud");
+        }
+    }
+
+    private void checkRemoteParam(SMSModemRemoteBridgeConfiguration config) throws ModemConfigurationException {
+        try {
+            InetAddress inetAddress = InetAddress.getByName(config.ip);
+            String ip = inetAddress.getHostAddress();
+
+            // test reachable address :
+            try (Socket s = new Socket(ip, config.networkPort)) {
+            }
+
+        } catch (IOException | NumberFormatException ex) {
+            // no ip
+            throw new ModemConfigurationException(
+                    config.ip + ":" + config.networkPort + " is not a reachable address:port", ex);
+        }
+    }
+
+    private String resolveEventualSymbolicLink(String serialPortOrIp) {
+        String keepResult = serialPortOrIp;
+        Path maybePath = Paths.get(serialPortOrIp);
+        File maybeFile = maybePath.toFile();
+        if (maybeFile.exists() && Files.isSymbolicLink(maybePath)) {
+            try {
+                maybePath = maybePath.toRealPath();
+                keepResult = maybePath.toAbsolutePath().toString();
+            } catch (IOException e) {
+            } // nothing to do, not a valid symbolic link, return
+        }
+        return keepResult;
+    }
+
+    public boolean isRunning() {
+        return modem != null && (modem.getStatus() == Status.Started || modem.getStatus() == Status.Starting);
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        if (childHandler instanceof SMSConversationHandler) {
+            childHandlers.add((SMSConversationHandler) childHandler);
+        } else {
+            logger.error("The SMSModemBridgeHandler can only handle SMSConversation as child");
+        }
+    }
+
+    @Override
+    public void childHandlerDisposed(ThingHandler childHandler, Thing childThing) {
+        childHandlers.remove(childHandler);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void messageReceived(InboundMessage message) {
+        String sender = message.getOriginatorAddress().getAddress();
+        Payload payload = message.getPayload();
+        String messageText;
+        if (payload.getType().equals(Type.Text)) {
+            String text = payload.getText();
+            if (text != null) {
+                messageText = text;
+            } else {
+                logger.error("Message has no payload !");
```

Comment:
See above. Logging to error. Please check all.

Reply:
Done